packages/quantic/force-app/main/default/lwc/quanticHeadlessLoader/quanticHeadlessLoader.js

```diff
@@ -94,35 +104,41 @@ const instantiateWindowEngineObject = (element, engineId) => {
  */
 async function initEngine(engineId) {
   try {
+    if (window.coveoHeadless[engineId].bindings.engine) {
+      throw new Error(`Engine already instantiated for engine ID: ${engineId}`);
+    }
+    if (!window.coveoHeadless[engineId].options) {
+      throw new Error('Engine options have not been set.');
+    }
     if (dependencyPromises.length === 0) {
       throw new Error('Dependencies were never requested.');
     }
     await Promise.all(dependencyPromises);
 
-    if (!window.coveoHeadless[engineId].config) {
-      throw new Error('Engine configuration has not been set.');
-    }
-    const configuration = await window.coveoHeadless[engineId].config.promise;
-    return CoveoHeadless.buildSearchEngine({configuration})
+    const options = await window.coveoHeadless[engineId].options.promise;
+    return CoveoHeadless[window.coveoHeadless[engineId].engineConstructor](options);
```

Comment:
If my understanding is correct, it means one could not pass its own engine constructor function. It has to be defined in `CoveoHeadless`. Isn't it preventing a customer from extending our engines?